### DISCUSSION TITLE
not booting properly in incomplete IPv6 environment

### DISCUSSION THREAD

**Discussion by n3gwg** [ipxe/ipxe#957] [Q&A] [answered]
iPXE users, developers, et alia:

I am running pfSense under ProxMox and also trying to PXE boot netboot.xyz in another VM under ProxMox. The environment seems to be serving up the netboot.ipxe.efi file properly (as it does start up). However, the foregoing notwithstanding, I am having problems getting the chain booting to occur properly. My TFTP server offers up the menu.ipxe file as depicted herein below (it is executing properly also), though for some reason DNS service seems only to provide an IPv6 IP address for boot.netboot.xyz and my network uses IPv4 exclusively, thus it never can access the link.

Can someone suggest some ideas as to how I can work around this problem so that I will obtain IPv4 DNS addresses in pursuance of resolving the URL properly and chain loading the menu as would normally be expected to occur. In fact, I tried to perform two nslookup commands below in order to assure it could resolve IPv4 DNS resolution (and it does, as foxnews.com shows an IPv4 IP address but boot.netboot.xyz gets an IPv6 IP address).

One such solution would be for netboot.xyz folks to create a "bootv4.netboot.xyz" DNS entry that only had IPv4 IP addresses associated with it, but I'd prefer to resolve the problem absent requesting that they reconfigure their networking.

My menu.ipxe is instantiated as follows hereupon:

#!ipxe
isset ${net0/mac} && echo MAC address: ${net0/mac} || echo No MAC address
isset ${net0/ip} && echo IP address: ${net0/ip} || echo No IP address
isset ${net0/dns} && echo DNS address: ${net0/dns} || echo No DNS address
nslookup address boot.netboot.xyz
echo DNS address: boot.netboot.xyz: ${address}
nslookup address www.foxnews.com
echo DNS address: www.foxnews.com : ${address}
set conn_type https
chain --autofree https://boot.netboot.xyz/menu.ipxe || echo FAILED: HTTPS - Attempting HTTP...
set conn_type http
chain --autofree http://boot.netboot.xyz/menu.ipxe  || echo FAILED: HTTP  - Dropping to command line
shell

![Screenshot from 2023-05-27 22-09-33](https://github.com/ipxe/ipxe/assets/57764892/8c710655-78cd-4da7-b878-54ae62b56345)

**Comment by NiKiZe** [comment]
Boot into iPXE to get a prompt, check config command to verify you don't have any v6 address.
DNS6 should only be used if you have a v6 route.
I would try and fix the network to have working v6

But it would be nice if iPXE could have RFC6555 implemented.

**Reply by n3gwg** [reply]
Nikize,

At the very end of my menu.ipxe script it ends up in a shell if all else fails (and so all else did fail and I did get an IPXE prompt. I tried the config command and it still is getting IPv6 DHCP'ed somehow (I will have to look with Wireshark and see what is sending it), but honestly I do not want IPv6 on my network and I do not use it. I just want netboot.xyz to boot properly and to honor IPv4 addressing and nothing else.

The point of the matter is, if I am DHCPing IPv4 and IPv6, why is it constantly trying to use IPv6 instead of IPv4?

Stuart

**Reply by NiKiZe** [reply]
Sorry I missed that you already had the prompt.
The good thing is that you found why iPXE does try IPv6.
If you don't want IPv6 to be used, then you need to ensure there is nothing giving out an IPv6 route.
Why would you not want proper IPv6? I understand that you don't want missconfigured IPv6 setup, unless for testing. But why ignore the current standard internet protocol? (RFC8200)

By RFC spec, IPv6 should be preferred if available. If it is available in your setup, it should use that.
The big question should be, why don't everyone have a working IPv6 setup by now 😉
Would recommend learning how it works instead of trying and ignore it.

**Comment by n3gwg** [comment]
The bottom line is that I have a 192.168.0.0/16 network which is fine for my needs, I do not need IPv6 or the confusion of it. Moreover, my ISP supports IPv4 currently, so I do not want it. How do I disable it in IPXE? This is what I am attempting to do. I should not have to go troubleshooting my entire network, why  can I not just tell IPXE to not use it? I am not familiar with troubleshooting it and do not wish to learn that just to use IPXE. I just want IPXE to stop trying to force me to use something I have no desire to have on my network. I am not here to have a political discussion over IPv6, I just want it disabled.

I guess if there is no way to disabled it in IPXE and it forces it on my network, I will have to not use IPXE. That is a shame as it seems a really nice product.

Thanks.
Stuart

**Reply by NiKiZe** [reply]
The fact is, your network says it has IPv6, the spec say to use IPv6 if it exists
Sure, you can rebuild iPXE without IPv6 support, but it sounds like the easiest solution for you is to have your network configured correctly.

**Comment by mcb30** [comment, marked as answer]
iPXE will attempt to perform IPv6 (AAAA record) resolution of DNS names only if you provide it with an IPv6 DNS server address.  (This is a heuristic that we use as a guess for "is the user trying to use IPv6?").

You can therefore prevent iPXE from attempting IPv6 DNS resolution by making sure that you give it only an IPv4 DNS server address.  The simplest way to do this is to perform only IPv4 network configuration in the first place, e.g. using `ifconf -c dhcp` (see https://ipxe.org/cmd/ifconf).  Alternatively, since your network does not provide a fully functional IPv6 environment, then disable the IPv6 router advertisements within your network.

**Reply by stappersg** [reply]
Suggestion for subject title change, in `sed` syntax:
```sed
s/not booting properly/not booting properly in incomplete IPv6 environment/
```

**Reply by n3gwg** [reply]
I have two APs that are there as WiFi entry points onto my network and not really functioning as routers (pfSense does my routing and network services). The APs have WiFi roaming setup as well and it works very well (the Linksys routers/APs run OpenWRT). I went and disabled IPv6 network on those APs as well and it seems to work better now and is booting into the menus for netboot.xyz.

Thanks for everyone's help.

One last point, I am not against IPv6 so much as I have not had need or time to deploy it.
